src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ImportStatusBauService.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Services;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Interfaces.Security;
+using GovUk.Education.ExploreEducationStatistics.Data.Processor.Model;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Cosmos.Table;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using static GovUk.Education.ExploreEducationStatistics.Common.TableStorageTableNames;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class ImportStatusBauService : IImportStatusBauService
+    {
+        private readonly ITableStorageService _tableStorageService;
+        private readonly ILogger _logger;
+        private readonly IUserService _userService;
+
+        public ImportStatusBauService(ITableStorageService tableStorageService,
+            ILogger<ImportStatusBauService> logger,
+            IUserService userService)
+        {
+            _tableStorageService = tableStorageService;
+            _logger = logger;
+            _userService = userService;
+        }
+
+        public async Task<Either<ActionResult, List<ImportStatusBau>>> GetAllIncompleteImports()
+        {
+            return await _userService
+                .CheckCanViewAllImports()
+                .OnSuccess(async () =>
+                {
+                    var queryFilterCondition = TableQuery.GenerateFilterCondition("Status",
+                        QueryComparisons.NotEqual,
+                        IStatus.COMPLETE.ToString());
+                    var tableQuery = new TableQuery<DatafileImport>().Where(queryFilterCondition);
+                    var queryResults = await _tableStorageService.ExecuteQueryAsync(
+                        DatafileImportsTableName, tableQuery);
+
+                    return queryResults.Where(result =>
+                    {
+                        if (result.Message == "")
+                        {
+                            _logger.LogWarning(
+                                $"{MethodBase.GetCurrentMethod()} received Imports row with empty Message. " +
+                                $"PartitionKey: \"{result.PartitionKey}\", RowKey: \"{result.RowKey}\"");
+                            return false;
+                        }
+                        return true;
+                    }).Select(result =>
+                    {
+                        var m = JsonConvert.DeserializeObject<ImportMessage>(result.Message);
```

Comment:
Thinking about it some more, I have a few reservations about this as it could lead to the frontend displaying stale information about the release/topic/publication. 

These details can be changed in the admin, so it's not out of the realms of possibility that they would be incorrect if an import has been sat in the failed status for some time. Now the question would be, is this acceptable? I'm potentially erring on yes as it's just a BAU screen, but it could be a little annoying if this trips us up in the future when trying to debug a problem.

The alternative would be to actually retrieve the up-to-date release from the database and use that in the view model instead. A lot will depend on what we're actually planning to display in the frontend. What do you think?

Reply:
My thought is leave it until its merged to dev - and possibly even until the frontend work is done - as it'll then be much easier to see if this is an issue, and more efficient. And as you say, it's a BAU tool, so if this ends up being an issue, it's not a pressing concern anyway.

Reply:
Coolio, we'll go with that :+1: